tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,794 @@
+        # # Start it in a clean node
+        # consumer.start()
+
+        # Producer
+        producer = self._create_producer()
+
+        # Prepare to save timigs
+        iteration_timings = {
+            "high_watermark": {},
+            "produce": {},
+            "manifest_uploads": {},
+            "segment_uploads": {},
+            "sp_mn_uploads_total": {},
+            "consumed_messages": {},
+            "consume": {},
+            "checks": {},
+            "scrub": {},
+            "restart": {}
+        }
+
+        # Announce scrub_exception
+        scrub_exception = None
+        # Do iterations of producing messages
+        while iteration <= iteration_count:
+            self.logger.info(f"Starting iteration {iteration},"
+                             f"runtime {self.params.target_runtime}s")
+            # Consumer are to be created each time and released at the end
+
+            # rand_consumer = self._create_rnd_consumer()
+            # consumer = self._create_seq_consumer()
+            # consumer = self._create_group_consumer()
+
+            # Producer
+            produce_start = time.time()
+            producer.start()
+            # wait for messages to appear in queue
+            producer.wait_for_acks(10000, timeout_sec=60, backoff_sec=10)
+            producer.wait_for_offset_map()
+            # wait and stop
+            # producer.wait(timeout_sec=self.params.expected_producer_duration)
+            # produce_duration = time.time() - produce_start
+            # producer.stop()
+            # Timing
+
+            # Debug consumer stage
+            # Consumer will use all available messages on loop
+            # Generating service load in the process
+            consume_start = time.time()
+            # It is important to start clean consumer
+            consumer.start()
+            # wait for all desired messages to be produced
+            producer.wait(timeout_sec=self.params.expected_producer_duration)
+            produce_duration = time.time() - produce_start
+            self.logger.info("Done producing messages")
+            # wait for consumer to hit end of the queue
+            consumer.wait(timeout_sec=self.params.max_consumer_duration)
+            consume_duration = time.time() - consume_start
+            self.logger.info("Done consuming messages")
+            # Stop everything
+            producer.stop()
+            consumer.stop()
+
+            # Dict for all checks
+            chk = RedpandaChecksBase()
+            checks_start = time.time()
+            # Calculate actual bitrate
+            actual_byte_rate = (self.params.write_bytes / produce_duration)
+            mbps = int(actual_byte_rate / (1024 * 1024))
+            self.logger.info(f"Produced {self.params.write_bytes} "
+                             f"in {produce_duration}s, {mbps}MiB/s")
+            # Add it for checking
+            chk.add_arg(f"i{iteration}_actual_byte_rate", actual_byte_rate)
```

Comment:
The quantity of checks makes it kind of hard to read through the test at a high level (i.e. to see what steps it is taking in terms of produce, consume, etc).  Perhaps it makes sense to move the definition of all the checks somewhere else, and then be able to write the body of the test as a briefer series of actions, with one-line calls into the predefined checks at the relevant points.